docs/source/var-handlers.rst

```diff
@@ -0,0 +1,155 @@
+            table: CMIP6_Amon.json
+            unit_conversion: null
+            formula: (PRECC + PRECL) * 1000.0
+            positive: null
+            levels: null
+
+2. ``e3sm_to_cmip/cmor_handlers/vars`` directory
+    - Each handler is defined as Python module (``.py`` file). This is the legacy design for  defining handlers, which is progressively being be replaced by ``handlers.yaml``.
+    - ``/vars`` stores *legacy* **atmosphere land variable handlers** including: ``areacella``, ``clisccp``, ``orog``, ``pfull``, ``phalf``)
+    - These handlers need to be refactored since they still depend on CDAT modules or contain redudant code that has since been generalized into helper functions. **DO NOT add any new handlers in this directory.** Instead, add new handlers to ``handlers.yaml``.
+
+3. ``e3sm_to_cmip/cmor_handlers/mpas_vars`` directory
+    - ``/mpas_vars`` **stores handler definitions for MPAS ocean and sea-ice variable handlers.** MPAS variables require additional processing requirements (e.g., use of mesh files).
+    - The development team is considering refactoring the design of these handlers.
+
+How to add new atmosphere and land variable handlers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Append a new entry to the ``e3sm_to_cmip/cmor_handlers/handlers.yaml`` file.
+2. If the handler has a ``formula``, add a formula function to ``e3sm_to_cmip/cmor_handlers/_formulas.py``.
+
+  - The function parameters must include ``data`` (a dictionary mapping variables to its underlying ``np.ndarray``) and ``index`` (the index within the array to apply the formula to).
+  - Example:
+
+    .. code-block:: python
+
+        def cLitter(data: Dict[str, np.ndarray], index: int) -> np.ndarray:
+            """
+            cLitter = (TOTLITC + CWDC)/1000.0
+            """
+            outdata = (data["TOTLITC"][index, :] + data["CWDC"][index, :]) / 1000.0
+
+            return outdata
+
+How to add new MPAS ocean and sea-ice variable handlers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Adding a variable handler for MPAS variables is slightly more involved process.
+
+You need to create a Python module in ``/cmor_handlers/mpas_vars``. We recommend taking a look
+at the existing modules such as ``so.py`` to get idea on how to add an MPAS handler.
+
+How ``e3sm_to_cmip`` derives handlers for variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``e3sm_to_cmip`` **derives the appropriate variable handlers to use based on the available E3SM variables in the input datasets.** Afterwards, it applies any necessary unit conversions, formulas, etc. during the CMORizng process.
+
+For example, let's say we want to CMORize the variable ``"pr"`` and we pass an E3SM input dataset that has the variables ``"PRECC"`` and ``"PRECL"``. ``e3sm_to_cmip`` derives the appropriate ``"pr"`` variable handler using this logic flow:
+
+1. Run ``e3sm_to_cmip --var-list pr --input-path <SOME_INPUT_PATH>``
```

Comment:
I'm a little bit confused about type of --var-list as input, in the helper script it says`Space separated list of variables`, but I saw instances of using, pr, 'pr', ['pr'], I suppose all three are supported as input?

Reply:
``--varlist`` accepts a list of strings with or without a comma (`pr, prw` or `pr prw` ). I prefer with a comma though because it looks a cleaner.